ggu/main.c

```diff
@@ -591,60 +591,74 @@ submit_ggu(struct rte_mbuf **pkts, unsigned int num_pkts,
 	return ret;
 }
 
+static inline void
+process_back_nic(struct ggu_config *ggu_conf,
+	uint16_t port_in, uint16_t rx_queue, uint16_t max_pkt_burst)
+{
+	struct rte_mbuf *bufs[max_pkt_burst];
+	uint16_t num_rx = rte_eth_rx_burst(port_in, rx_queue,
+		bufs, max_pkt_burst);
+	unsigned int i;
+
+	if (unlikely(num_rx == 0))
+		return;
+
+	for (i = 0; i < num_rx; i++)
+		process_single_packet(bufs[i], ggu_conf);
+}
+
+static inline void
+process_mb(struct ggu_config *ggu_conf)
+{
+	unsigned int mailbox_burst_size = ggu_conf->mailbox_burst_size;
+	struct ggu_request *reqs[mailbox_burst_size];
+	unsigned int num_reqs = mb_dequeue_burst(&ggu_conf->mailbox,
+		(void **)reqs, mailbox_burst_size);
+	unsigned int i;
+
+	if (unlikely(num_reqs == 0))
+		return;
+
+	for (i = 0; i < num_reqs; i++) {
+		unsigned int j;
+		for (j = 0; j < reqs[i]->num_pkts; j++) {
+			process_single_packet(reqs[i]->pkts[j], ggu_conf);
+		}
+	}
+
+	mb_free_entry_bulk(&ggu_conf->mailbox, (void * const *)reqs, num_reqs);
+}
+
 static int
 ggu_proc(void *arg)
 {
 	uint32_t lcore = rte_lcore_id();
 	struct ggu_config *ggu_conf = (struct ggu_config *)arg;
 	uint16_t port_in = ggu_conf->net->back.id;
 	uint16_t rx_queue = ggu_conf->rx_queue_back;
-	unsigned int i;
 	uint16_t max_pkt_burst = ggu_conf->max_pkt_burst;
 
-	GGU_LOG(NOTICE, "The GK-GT unit is running at lcore = %u\n", lcore);
+	GGU_LOG(NOTICE, "The GT-GK unit is running at lcore = %u\n", lcore);
 
 	/*
-	 * Load a set of GK-GT packets from the back NIC
+	 * Load sets of GT-GK packets from the back NIC
 	 * or from the GGU mailbox.
 	 */
-	if (hw_filter_ntuple_available(&ggu_conf->net->back)) {
-		while (likely(!exiting)) {
-			struct rte_mbuf *bufs[max_pkt_burst];
-			uint16_t num_rx = rte_eth_rx_burst(port_in, rx_queue,
-				bufs, max_pkt_burst);
-
-			if (unlikely(num_rx == 0))
-				continue;
-
-			for (i = 0; i < num_rx; i++)
-				process_single_packet(bufs[i], ggu_conf);
+	bool acl_rx_enabled = acl_enabled(&ggu_conf->net->back);
+	bool ntuple_filter_enabled =
+		hw_filter_ntuple_enabled(&ggu_conf->net->back);
+	while (likely(!exiting)) {
+		if (ntuple_filter_enabled) {
```

Comment:
Style: `ntuple_filter_enabled` is used first, so move its declaration before the declaration of `acl_rx_enabled`.